File: native-android/app/src/main/java/com/ronecaplaytv/nativeapp/activation/SupportProfile.kt
package com.ronecaplaytv.nativeapp.activation

import java.net.URI

enum class SupportProfileSource {
    Seller,
    System,
    Generic,
}

data class SupportProfile(
    val source: SupportProfileSource = SupportProfileSource.Generic,
    val displayName: String = "Suporte",
    val whatsapp: String? = null,
    val email: String? = null,
    val supportText: String? = "Envie este código ao seu fornecedor.",
    val businessHours: String? = null,
    val contactUrl: String? = null,
    val showInApp: Boolean = true,
) {
    val primaryContactUri: String?
        get() = SupportContactPolicy.primaryUri(this)

    val contactLabel: String?
        get() = when {
            SupportContactPolicy.safeHttpsUri(contactUrl) != null -> "Abrir atendimento"
            SupportContactPolicy.safeWhatsappUri(whatsapp) != null -> "Abrir WhatsApp"
            SupportContactPolicy.safeEmailUri(email) != null -> "Enviar e-mail"
            else -> null
        }

    companion object {
        fun generic() = SupportProfile()
    }
}

/**
 * Política pura e testável para contatos recebidos do backend. O aplicativo
 * nunca aceita esquemas arbitrários nem URLs HTTPS com credenciais embutidas.
 */
object SupportContactPolicy {
    private val emailPattern = Regex("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$")

    fun primaryUri(profile: SupportProfile): String? =
        safeHttpsUri(profile.contactUrl)
            ?: safeWhatsappUri(profile.whatsapp)
            ?: safeEmailUri(profile.email)

    fun safeHttpsUri(value: String?): String? {
        val raw = value?.trim()?.takeIf { it.length in 1..2048 } ?: return null
        val parsed = runCatching { URI(raw) }.getOrNull() ?: return null
        if (!parsed.scheme.equals("https", ignoreCase = true)) return null
        if (parsed.host.isNullOrBlank() || parsed.userInfo != null) return null
        return parsed.normalize().toASCIIString()
    }

    fun safeWhatsappUri(value: String?): String? {
        val raw = value?.trim()?.takeIf { it.length <= 40 } ?: return null
        val normalized = raw.replace(Regex("[^\\d+]"), "")
        if (!Regex("^\\+?\\d{8,15}$").matches(normalized)) return null
        return "https://wa.me/${normalized.filter(Char::isDigit)}"
    }

    fun safeEmailUri(value: String?): String? {
        val normalized = value?.trim()?.lowercase()?.takeIf { it.length <= 254 } ?: return null
        if (!emailPattern.matches(normalized)) return null
        return "mailto:$normalized"
    }
}

File: smart-tv/src/supportProfile.ts
export type SupportProfileSource = "seller" | "system" | "generic";

export interface SupportProfile {
  source: SupportProfileSource;
  displayName: string;
  whatsapp: string | null;
  email: string | null;
  supportText: string | null;
  businessHours: string | null;
  contactUrl: string | null;
  showInApp: boolean;
  primaryContactUri: string | null;
  contactLabel: string | null;
}

export function genericSupportProfile(): SupportProfile {
  return {
    source: "generic",
    displayName: "Suporte",
    whatsapp: null,
    email: null,
    supportText: "Envie este código ao seu fornecedor.",
    businessHours: null,
    contactUrl: null,
    showInApp: true,
    primaryContactUri: null,
    contactLabel: null
  };
}

function safeString(value: unknown, limit: number): string | null {
  const text = String(value ?? "").trim();
  return text && text.length <= limit ? text : null;
}

function safeHttps(value: unknown): string | null {
  const raw = safeString(value, 2048);
  if (!raw) return null;
  try {
    const url = new URL(raw);
    if (url.protocol !== "https:" || !url.hostname || url.username || url.password) return null;
    url.hash = "";
    return url.toString();
  } catch { return null; }
}

function safeWhatsapp(value: unknown): { display: string; uri: string } | null {
  const raw = safeString(value, 40);
  if (!raw) return null;
  const normalized = raw.replace(/[^\d+]/g, "");
  if (!/^\+?\d{8,15}$/.test(normalized)) return null;
  return { display: raw, uri: `https://wa.me/${normalized.replace(/\D/g, "")}` };
}

function safeEmail(value: unknown): { display: string; uri: string } | null {
  const raw = safeString(value, 254)?.toLowerCase();
  if (!raw || !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(raw)) return null;
  return { display: raw, uri: `mailto:${raw}` };
}

export function normalizeSupportProfile(value: unknown): SupportProfile {
  if (!value || typeof value !== "object") return genericSupportProfile();
  const raw = value as Record<string, unknown>;
  if (raw.showInApp === false) return genericSupportProfile();
  const https = safeHttps(raw.contactUrl);
  const whatsapp = safeWhatsapp(raw.whatsapp);
  const email = safeEmail(raw.email);
  const primaryContactUri = https || whatsapp?.uri || email?.uri || null;
  const source = raw.source === "seller" || raw.source === "system" ? raw.source : "generic";
  return {
    source,
    displayName: safeString(raw.displayName, 120) || "Suporte",
    whatsapp: whatsapp?.display || null,
    email: email?.display || null,
    supportText: safeString(raw.supportText, 280),
    businessHours: safeString(raw.businessHours, 160),
    contactUrl: https,
    showInApp: true,
    primaryContactUri,
    contactLabel: https ? "Abrir atendimento" : whatsapp ? "Abrir WhatsApp" : email ? "Enviar e-mail" : null
  };
}
